sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -427,7 +473,14 @@ private[hive] class HiveClientImpl(
   }
 
   override def alterTable(tableName: String, table: CatalogTable): Unit = withHiveState {
-    val hiveTable = toHiveTable(table)
+    // getTableOption removes all the Hive-specific properties. Here, we fill them back to ensure
+    // these properties are still available to the others that share the same Hive metastore.
+    // If users explicitly alter these Hive-specific properties through ALTER TABLE DDL, we respect
+    // these user-specified values.
+    val hiveSpecificProps = table.ignoredProperties.filter {
+      case (key, _) => HiveStatisticsProperties.contains(key)
+    }
+    val hiveTable = toHiveTable(table.copy(properties = table.properties ++ hiveSpecificProps))
```

Comment:
`table.properties ++ hiveSpecificProps`, by doing this, we won't respect user-specified Hive-specific properties, which are in `table.properties`.